homeassistant/components/salt/device_tracker.py

```diff
@@ -0,0 +1,72 @@
+"""Support for Salt Fiber Box routers."""
+import logging
+
+from saltbox import RouterLoginException, RouterNotReachableException, SaltBox
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import (
+    DOMAIN,
+    PLATFORM_SCHEMA,
+    DeviceScanner,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def get_scanner(hass, config):
+    """Return the Salt device scanner."""
+    scanner = SaltDeviceScanner(config[DOMAIN])
+
+    # Test whether the router is accessible.
+    data = scanner.get_salt_data()
+    return scanner if data is not None else None
```

Comment:
```py
return scanner if data else None
```

Reply:
Is it actually necessary to do this check at all? If the component is faulty / misconfigured it will raise errors later. Is there a benefit to returning None in this case?

Reply:
Yes, we don't want to set up a platform that doesn't work.

Reply:
Makes sense, thanks :)

Reply:
I'm leaving the explicit `is not None` to distinguish between None (error) and an empty list (no devices connected).